tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -475,6 +515,28 @@ class TestType{{{methodSource}}}" );
 		return (compilation, methodDecl);
 	}
 
+
+	public static (Compilation, MethodDeclarationSyntax) ParseMethodWithIAsyncEnumerable( string methodSource ) {
+		var wrappedAndParsed = CSharpSyntaxTree.ParseText( @$"
+using System.Threading.Tasks;
+using System.Threading;
+using System.Collections.Generic;
+using D2L.CodeStyle.Annotations;
+
+class TestType{{
+{methodSource}async IAsyncEnumerable<string> MethodReturningIAsyncEnumerable() {{

```

Comment:
For completeness so that both cases above are valid code:

```suggestion
{methodSource}async IAsyncEnumerable<string> MethodReturningIAsyncEnumerable( int p = 0 ) {{
```